src/Bridge/Doctrine/Orm/Util/EagerLoadingTrait.php

```diff
@@ -62,30 +62,80 @@ private function getBooleanOperationAttribute(string $resourceClass, array $opti
     }
 
     /**
-     * Checks if the class has an associationMapping with FETCH=EAGER.
-     *
-     * @param array $checked array cache of tested metadata classes
+     * Checks if the class has a one-to-many associationMapping with FETCH=EAGER in the where clause.
      */
-    private function hasFetchEagerAssociation(EntityManagerInterface $em, ClassMetadataInfo $classMetadata, array &$checked = []): bool
+    private function hasFetchEagerOneToManyInWhere(QueryBuilder $queryBuilder, ClassMetadataInfo $classMetadata): bool
     {
-        $checked[] = $classMetadata->name;
-
-        foreach ($classMetadata->getAssociationMappings() as $mapping) {
-            if (ClassMetadataInfo::FETCH_EAGER === $mapping['fetch']) {
-                return true;
+        $em = $queryBuilder->getEntityManager();
+        $wherePart = $queryBuilder->getDQLPart('where');
+        $aliasMapping = $this->createAliasMapping($queryBuilder);
+        $checked = [];
+
+        foreach ($wherePart->getParts() as $part) {
+            foreach ($aliasMapping as $alias => $fieldPath) {
+                if ('' === $fieldPath) {
+                    // Root alias
+                    continue;
+                }
+
+                if (false !== strpos((string) $part, $alias.'.')) {
+                    $fields = explode('.', $fieldPath);
+                    $clonedClassMetadata = clone $classMetadata;
```

Comment:
Line 88 : it's assigned by the "targetEntity" meta data. But when we restart the loop, we need to take the original one. I will be glad if you have an ideal to avoid the clone. :)

Reply:
Yes, you still required to take back the original one at this moment, but a `$clonedClassMetadata = $classMetadata;` is enough because you do not modify the original one.

Reply:
`$classMetadata` is an object. Object's assignments are [always by reference](https://www.php.net/manual/en/language.oop5.references.php).

Reply:
You are totally right. I didn't read carefully your comment. I will remove the clone. Thank you.